docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -577,6 +577,58 @@ Person.find("name = :name and status = :status",
 
 Every query operation accepts passing parameters by index (`Object...`), or by name (`Map<String,Object>` or `Parameters`).
 
+=== Query projection
+
+Query projection can be done with the `project(Class)` method on the `PanacheQuery` object that is returned by the `find()` methods.
+
+You can use it to restrict which fields will be returned by the database.
+
+Hibernate will use **DTO projection** and generate a SELECT clause with the attributes from the projection class.
+This is also called **dynamics instanciation** or **constructor expression**, more info can be found on the Hibernate guide:
+link:https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#hql-select-clause[hql select clause]
+
+The projection class needs to be a valid Java Bean and have a constructor that contains all its attributes, this constructor will be used to
+instantiate the projection DTO instead of using the entity class. This must be the only constructor of the class.
```

Comment:
Why? What happens if there are more?

Reply:
I took the first one.
If there are more, we will need a way to know which one to choose to build the select clause.
We can do it by adding an annotation to denote the correct constructor, but I prefere to keep it simple for the moment and only allow one constructor.
I can add a check in the code and fail if there is more than one constructor
We can revisit this later if needed.

Reply:
OK fine.